eis_toolkit/exploratory_analyses/k_means_cluster.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import KMeans
+from sklearn.preprocessing import StandardScaler
+
+from eis_toolkit.exceptions import EmptyDataFrameException
+
+
+def _k_means_clustering(
+    data: gdp.GeoDataFrame, number_of_clusters: int, random_state: Optional[int] = None
+) -> gdp.GeoDataFrame:
+
+    coordinates = data.geometry.apply(lambda geom: [geom.x, geom.y])
+    coordinates_array = np.array(coordinates.tolist())
+    standardized_data = StandardScaler().fit_transform(coordinates_array)
+
+    kmeans = KMeans(n_clusters=number_of_clusters, random_state=random_state)
+    kmeans.fit(standardized_data)
+
+    data["cluster"] = kmeans.labels_
+
```

Comment:
Now that I think about this, we could also add a column that has 1 if the point is a cluster centroid and 0 if it is not a centroid.